homeassistant/components/russound_rio/config_flow.py

```diff
@@ -0,0 +1,107 @@
+"""Config flow to configure russound_rio component."""
+
+from __future__ import annotations
+
+import asyncio
+import logging
+from typing import Any
+
+from aiorussound import Russound
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigFlow, ConfigFlowResult
+from homeassistant.const import CONF_HOST, CONF_MODEL, CONF_PORT
+from homeassistant.helpers import config_validation as cv
+
+from .const import DOMAIN, RUSSOUND_RIO_EXCEPTIONS, NoPrimaryControllerException
+
+DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required(CONF_HOST): cv.string,
+        vol.Optional(CONF_PORT, default=9621): cv.port,
+    }
+)
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def find_primary_controller_metadata(
+    controllers: list[tuple[int, str, str]],
+) -> tuple[str, str]:
+    """Find the mac address of the primary Russound controller."""
+    for controller_id, mac_address, controller_type in controllers:
+        # The integration only cares about the primary controller linked by IP and not any downstream controllers
+        if controller_id == 1:
+            return (mac_address, controller_type)
+    raise NoPrimaryControllerException
+
+
+class FlowHandler(ConfigFlow, domain=DOMAIN):
+    """Russound RIO configuration flow."""
+
+    VERSION = 1
+
+    async def async_step_user(
+        self, user_input: dict[str, Any] | None = None
+    ) -> ConfigFlowResult:
+        """Handle a flow initialized by the user."""
+        errors: dict[str, str] = {}
+        if user_input is not None:
+            host = user_input[CONF_HOST]
+            port = user_input[CONF_PORT]
+
+            controllers = None
+            russ = Russound(self.hass.loop, host, port)
+            try:
+                async with asyncio.timeout(5):
+                    await russ.connect()
+                    controllers = await russ.enumerate_controllers()
+                    metadata = find_primary_controller_metadata(controllers)
+                    await russ.close()
+            except RUSSOUND_RIO_EXCEPTIONS:
+                _LOGGER.exception("Could not connect to Russound RIO")
+                errors["base"] = "cannot_connect"
+            except NoPrimaryControllerException:
+                _LOGGER.exception(
+                    "Russound RIO device doesn't have a primary controller",
+                )
+                errors["base"] = "no_primary_controller"
+            else:
+                await self.async_set_unique_id(metadata[0])
+                data = {CONF_HOST: host, CONF_PORT: port, CONF_MODEL: metadata[1]}
```

Comment:
Not sure why we now store the model as well

Reply:
I'm likely misunderstanding the data field here. What's the intention for this feature? 
I'm trying to store the model (and honestly the MAC for the matter) to be used later when the device attributes are implemented in the future.

Reply:
Usually it's data use to connect to the device. Like unless its a very big hassle to fetch the model on runtime, we'd rather get it dynamically.

Reply:
Completely makes sense. I'll remove the model from data. It's not that big of a deal to grab @ runtime.